espressif/vscode-esp-idf-extension
# Feature/web extension

## Description

Add `ESP-IDF: Flash with WebSerial` to flash using esptool-js in web environments.

## Type of change

- New feature (non-breaking change which adds functionality)

## Steps to test this pull request

Provide a list of steps to test changes in this PR and required output

1. Click on `ESP-IDF: Flash with WebSerial` using the command palette
2. Choose the serial port
3. See flash progress and done result.

- Expected behaviour:

- Expected output:

A list of serial ports is shown to flash the program. Later a progress notification showing the flashing progress and done notification. The program should be working on the device.

## How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- Test A
- Test B

**Test Configuration**:
* ESP-IDF Version: 5.1
* OS (Windows,Linux and macOS): macOS but should run on a web interface like Github Codespaces.

## Dependent components impacted by this PR:

- Component 1
- Component 2

## Checklist
